File: src/1086-High-Five/1086.py
from collections import defaultdict
class Solution:
    def highFive(self, items: List[List[int]]) -> List[List[int]]:
        data = defaultdict(list)
        for it in items:
            data[it[0]].append(it[1])
        
        res = list()
        for it in data.items():
            res.append([it[0], sum(sorted(it[1], reverse=True)[:5])//5])
        return res

File: src/1086-High-Five/1086.cpp
class Solution 
{
public:
    vector<vector<int>> highFive(vector<vector<int>>& items) 
    {
        map<int, vector<int>> m;
        for (auto& it : items) m[it[0]].push_back(it[1]);
        vector<vector<int>> res;
        for (auto& it : m) 
        {
            sort(it.second.begin(), it.second.end(), greater<int>());
            res.push_back({it.first, accumulate(it.second.begin(), it.second.begin()+5, 0)/5});
        }
        return res;
    }
};